### ISSUE TITLE
A few npm install warnings from Travis

### ISSUE THREAD

**Issue by pdehaan** [mozilla/web-ext#486]
Only because I'm staring at the Travis logs waiting for my build to fail...

```
npm WARN deprecated babel-istanbul-loader@0.1.0: This package is deprecated in favor of istanbuljs/babel-plugin-istanbul
npm WARN peerDependencies The peer dependency eslint@>=2.0.0 included from eslint-plugin-flowtype will no
npm WARN peerDependencies longer be automatically installed to fulfill the peerDependency 
npm WARN peerDependencies in npm 3+. Your application will need to depend on it explicitly.
npm WARN optional dep failed, continuing fsevents@1.0.14
npm WARN deprecated tough-cookie@2.2.2: ReDoS vulnerability parsing Set-Cookie https://nodesecurity.io/advisories/130
npm WARN deprecated minimatch@2.0.10: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
```

I'd have to dig through each warning to see if there is anything we can do directly, but I think at the very least we should investigate the **babel-istanbul-loader@0.1.0** warning since it looks like that is a direct package.json devDependency and we could "easily" switch over to the recommended [**babel-plugin-istanbul**](https://www.npmjs.com/package/babel-plugin-istanbul) module.

**Comment by kumar303** [comment]
Yeah, some of these may be easier to fix than others (the direct deps). It would be good to reduce noise and fix them though.

**Comment by kumar303** [comment]
> switch over to the recommended babel-plugin-istanbul module

I think it should be straight forward. Someone did it on [addons-frontend](https://github.com/mozilla/addons-frontend/blob/master/package.json#L195) so you could possibly dig for the original patch.

**Comment by pdehaan** [comment]
if we ignore `peerDependencies` and `optional dep` warnings, we're left with:

- **babel-istanbul-loader**
- **tough-cookie**
- **minimatch**

---

```sh
$ npm ls babel-istanbul-loader

web-ext@1.4.0 /Users/pdehaan/dev/github/mozilla/web-ext
└── babel-istanbul-loader@0.1.0
```

```sh
$ npm ls tough-cookie@2.2.2

web-ext@1.4.0 /Users/pdehaan/dev/github/mozilla/web-ext
├─┬ sign-addon@0.1.3
│ └─┬ request@2.73.0
│   └── tough-cookie@2.2.2
│
└─┬ watchpack@1.1.0
  └─┬ chokidar@1.6.0
    └─┬ fsevents@1.0.14
      └─┬ node-pre-gyp@0.6.29
        └─┬ request@2.73.0
          └── tough-cookie@2.2.2
```

```sh
$ npm ls minimatch@2.0.10

web-ext@1.4.0 /Users/pdehaan/dev/github/mozilla/web-ext
└─┬ babel-istanbul@0.11.0
  └─┬ fileset@0.2.1
    └── minimatch@2.0.10
```

We're probably out of luck for **tough-cookie** and **minimatch** since those are pretty nested, and I'm not seeing any newer versions of modules in our package.json, apart from these (which seem to be unaffected):

```sh
$ npm outdated --depth 0

Package                 Current  Wanted  Latest  Location
addons-linter            0.15.1  0.15.1  0.15.3  web-ext
eslint-plugin-flowtype   2.16.1  2.16.1  2.17.1  web-ext
webpack-dev-server       1.15.1  1.15.1  1.15.2  web-ext
```

Re: **tough-cookie** and https://nodesecurity.io/advisories/130, it looks like that may be fixed in **tough-cookie@>=2.3.0**, which seems to be included in latest [**request@2.74.1**](https://github.com/request/request/blob/06e8f851ad460afc1094a622eec2b72f33489b3b/package.json#L44). I think that's already been fixed in latest **sign-addon** master, but just needs a new version published to npm with newer pinned **request** version number.

**UPDATE:** Submitted a PR to **babel-istanbul** to bump the **fileset** dependency version: https://github.com/jmcriffey/babel-istanbul/pull/73

**Comment by kumar303** [comment]
I pushed a release of sign-addon and web-ext master has it now as of https://github.com/mozilla/web-ext/pull/494

**Comment by wagnerand-moz** [comment]
Closing out ancient issue.
